include/AST/Types/StructTy.hpp

```diff
@@ -11,49 +11,85 @@
 
 namespace glu::types {
 
+struct Field {
+    std::string name;
```

Comment:
std::string allocates, so you'll still end up with a memory leak

Reply:
maybe use a Token, as struct field names should be identifiers that are written in the source?